certs/fake/certificate.go

```diff
@@ -8,100 +8,97 @@ import (
 	"crypto/x509/pkix"
 	"fmt"
 	"math/big"
-	"sync"
 	"time"
 
 	"github.com/zalando-incubator/kube-ingress-aws-controller/certs"
 )
 
-type caSingleton struct {
-	once      sync.Once
-	err       error
+type CA struct {
 	chainKey  *rsa.PrivateKey
 	roots     *x509.CertPool
 	chainCert *x509.Certificate
 }
 
-type CertificateProvider struct {
-	ca caSingleton
```

Comment:
This was something wrong. I can understand singleton CA but when it is a member its a singleton within CertificateProvider instance and then it was only used once...